Description: updating makefile
 updated makefile to make it more compatible with packaging
Author: Nicolas HOUDELOT (nicolas@demosdebs.org)
Last-Update: 2020-04-18
Applied-Upstream: no
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,20 @@
 #  The top-level makefile simply invokes all the other makefiles
-
-prefix = /usr/local
+SHELL = /bin/sh
+MAKE = make
+CC = gcc
+AR = ar
+prefix = /usr
+#variable de nettoyage
+RM_F = rm -f
 
 PROGRAMS = elfls objres rebind sstrip elftoc ebfc infect
 
+#variables d'instalation
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -m  755
+INSTALL_PROGRAM = $(INSTALL) -p    -m  755
+INSTALL_DATA    = $(INSTALL) -p    -m  644
+
 all: $(PROGRAMS)
 
 bin/%:
@@ -24,11 +35,11 @@
 infect: bin/infect doc/infect.1
 
 install: $(PROGRAMS)
-	mkdir -p $(prefix)/bin
-	cp bin/* $(prefix)/bin/.
-	mkdir -p $(prefix)/share/man/man1
-	cp doc/* $(prefix)/share/man/man1/.
+	$(INSTALL_DIR) $(DESTDIR)$(prefix)/bin/
+	$(INSTALL_PROGRAM) bin/*  $(DESTDIR)$(prefix)/bin/.
+	$(INSTALL_DIR) $(DESTDIR)$(prefix)/share/man/man1/
+	$(INSTALL_DATA) doc/*  $(DESTDIR)$(prefix)/share/man/man1/.
 
 clean:
 	for dir in elfrw $(PROGRAMS) ; do $(MAKE) -C$$dir clean ; done
-	rm -f $(PROGRAMS:%=bin/%) $(PROGRAMS:%=doc/%.1)
+	-@$(RM_F) $(PROGRAMS:%=bin/%) $(PROGRAMS:%=doc/%.1)
--- a/ebfc/Makefile
+++ b/ebfc/Makefile
@@ -1,7 +1,6 @@
 #  Makefile for ebfc
 
-CC = gcc
-CFLAGS = -Wall -Wextra -Wno-missing-field-initializers
+CFLAGS += -O3 -Wall -Wextra -pipe -Wno-missing-field-initializers
 
 # The main program.
 
@@ -15,7 +14,7 @@
 libelfparts.a: elfparts.o \
 	       ehdr.o phdrtab.o shdrtab.o progbits.o \
                strtab.o symtab.o hash.o rel.o got.o dynamic.o
-	ar crs libelfparts.a $^
+	$(AR) crs libelfparts.a $^
 
 elfparts.o: elfparts.c elfparts.h elfpartsi.h
 ehdr.o: ehdr.c elfparts.h elfpartsi.h
@@ -30,8 +29,8 @@
 dynamic.o: dynamic.c elfparts.h elfpartsi.h
 
 clean:
-	rm -f ebfc.o brainfuck.o
-	rm -f elfparts.o elfpseq.o
-	rm -f ehdr.o phdrtab.o shdrtab.o progbits.o strtab.o
-	rm -f symtab.o hash.o rel.o got.o dynamic.o
-	rm -f ebfc libelfparts.a
+	-@$(RM_F) ebfc.o brainfuck.o
+	-@$(RM_F) elfparts.o elfpseq.o
+	-@$(RM_F) ehdr.o phdrtab.o shdrtab.o progbits.o strtab.o
+	-@$(RM_F) symtab.o hash.o rel.o got.o dynamic.o
+	-@$(RM_F) ebfc libelfparts.a
--- a/elfls/Makefile
+++ b/elfls/Makefile
@@ -1,7 +1,6 @@
 #  Makefile for elfls
 
-CC = gcc
-CFLAGS = -Wall -Wextra -I../elfrw
+CFLAGS += -O3 -Wall -Wextra -pipe -I../elfrw
 
 elfls: elfls.c ../elfrw/libelfrw.a
 
@@ -9,4 +8,4 @@
 	$(MAKE) -C ../elfrw libelfrw.a
 
 clean:
-	rm -f elfls
+		-@$(RM_F) elfls
--- a/elfrw/Makefile
+++ b/elfrw/Makefile
@@ -1,13 +1,12 @@
 #  Makefile for libelfrw.a
 
-CC = gcc
-CFLAGS = -Wall -Wextra -O3
+CFLAGS += -O3 -Wall -Wextra -pipe
 
 LIBOBJS = elfrw.o elfrw_dyn.o elfrw_ehdr.o elfrw_phdr.o elfrw_rel.o \
           elfrw_shdr.o elfrw_sym.o elfrw_ver.o
 
 libelfrw.a: $(LIBOBJS)
-	ar crs $@ $^
+	$(AR) crs $@ $^
 
 elfrw.o: elfrw.c elfrw.h elfrw_int.h
 elfrw_dyn.o: elfrw_dyn.c elfrw_int.h
@@ -19,4 +18,4 @@
 elfrw_ver.o: elfrw_ver.c elfrw_int.h
 
 clean:
-	rm -f libelfrw.a $(LIBOBJS)
+	-@$(RM_F) libelfrw.a $(LIBOBJS)
--- a/elftoc/Makefile
+++ b/elftoc/Makefile
@@ -1,8 +1,7 @@
 #  Makefile for elftoc
 
-CC = gcc
-CFLAGS = -Wall -Wextra
-LDFLAGS = -Wall -Wextra
+CFLAGS += -O3 -Wall -Wextra -pipe
+LDFLAGS += -Wall -Wextra
 
 OBJS = gen.o names.o address.o readelf.o shdrtab.o phdrtab.o dynamic.o \
        pieces.o outbase.o outitems.o outelf64.o outelf32.o out.o elftoc.o
@@ -40,4 +39,4 @@
 mknames.o: mknames.c
 
 clean:
-	rm -f $(OBJS) elftoc elfnames.c mknames mknames.o
+	-@$(RM_F) $(OBJS) elftoc elfnames.c mknames mknames.o
--- a/infect/Makefile
+++ b/infect/Makefile
@@ -1,9 +1,8 @@
 #  Makefile for infect
 
-CC = gcc
-CFLAGS = -Wall -Wextra
+CFLAGS += -O3 -Wall -Wextra  -pipe
 
 infect: infect.c
 
 clean:
-	rm -f infect
+	-@$(RM_F) infect
--- a/objres/Makefile
+++ b/objres/Makefile
@@ -1,7 +1,7 @@
 #  Makefile for objres
 
 CC = gcc
-CFLAGS = -Wall -Wextra -I../elfrw
+CFLAGS += -O3 -Wall -Wextra -pipe -I../elfrw
 LDFLAGS = -Wall -Wextra
 
 objres: objres.o ../elfrw/libelfrw.a
@@ -12,4 +12,4 @@
 	$(MAKE) -C ../elfrw libelfrw.a
 
 clean:
-	rm -f objres objres.o
+	-@$(RM_F) objres objres.o
--- a/rebind/Makefile
+++ b/rebind/Makefile
@@ -1,7 +1,6 @@
 #  Makefile for rebind
 
-CC = gcc
-CFLAGS = -Wall -Wextra -I../elfrw
+CFLAGS += -O3 -Wall -Wextra -pipe -I../elfrw
 
 rebind: rebind.c ../elfrw/libelfrw.a
 
@@ -9,4 +8,4 @@
 	$(MAKE) -C ../elfrw libelfrw.a
 
 clean:
-	rm -f rebind
+	-@$(RM_F) rebind
--- a/sstrip/Makefile
+++ b/sstrip/Makefile
@@ -1,7 +1,6 @@
 #  Makefile for sstrip
 
-CC = gcc
-CFLAGS = -Wall -Wextra -I../elfrw
+CFLAGS += -O3 -Wall -Wextra -pipe -I../elfrw
 
 sstrip: sstrip.c ../elfrw/libelfrw.a
 
@@ -9,4 +8,4 @@
 	$(MAKE) -C ../elfrw libelfrw.a
 
 clean:
-	rm -f sstrip
+	-@$(RM_F) sstrip
